### PR TITLE
Step by Step Implementation Guide for K-Means Algorithm in R

### PR DESCRIPTION
**Pull request by lalith1403**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by lalith1403** [comment]
This fixes #358

**Comment by hectorkambow** [comment]
just making sure the images don't need sources 👍 @lalith1403

**Comment by lalith1403** [comment]
> just making sure the images don't need sources 👍 @lalith1403 

Yep. All are screenshots from my laptop.

File: articles/k-means-from-scratch-R/index.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+```
+
+To find the centroid of the newly formed cluster, we take the mean of all the points obtained above.
+The thinking is as follows: We need to find a point closest to all the cluster data points. Therefore, averaging the data points results in a point closest to the remaining points.
+
+```r
+mean(dataPoints[,1][c(total_distance[,1] <= total_distance[,2])])
+```
+
+We calculate the mean using the R function `mean.` This is an example on selecting elements conditionally that belong to a cluster and finding its centroid.
```
Comment (hectorkambow) [inline articles/k-means-from-scratch-R/index.md:225]: If we could clarify this sentence: 
"This is an example on selecting elements conditionally that belong to a cluster and finding its centroid."

I may have changed some things already - I just want to make sure things are clear to the audience. 
@lalith1403

Comment (hectorkambow) [reply articles/k-means-from-scratch-R/index.md]: See changes sentence - if that works ? 🚀

Comment (lalith1403) [reply articles/k-means-from-scratch-R/index.md]: Thank you for the updation. Yes it conveys the meaning.

**Comment by hectorkambow** [comment]
@kurt-sectionio 👋  Good afternoon Kurt - could you please review this K-means Algorithm in R article a tech review when you have a moment. 
It has a few code snippets. 
🚀 
Thank you

**Review by kurt-sectionio — APPROVED** [review]
Looks very good. Maybe you could mention that other distance measure can be used as well. Different distance measures contribute to the flexibility and wide applicability of K-means clustering.

**Pull request merged**